lib/manageiq/cross_repo/runner.rb

```diff
@@ -35,14 +36,25 @@ def run
 
     def run_tests
       with_test_env do
-        system!({"TRAVIS_BUILD_DIR" => test_repo.path.to_s}, "bash", "tools/ci/before_install.sh") if ENV["CI"] && File.exist?("tools/ci/before_install.sh")
-        system!(env_vars, "bin/setup")
-        system!(script_cmd)
+        require "yaml"
+        travis_yml = YAML.load_file(".travis.yml")
+
+        Array(travis_yml["before_install"]).each { |before_install| system!(env_vars, before_install) }
+        Array(travis_yml["install"]).each { |install| system!(env_vars, install) }
+        Array(travis_yml["before_script"]).each { |before_script| system!(env_vars, before_script) }
+
+        if script_cmd
+          system!(env_vars, script_cmd)
+        else
+          Array(travis_yml["script"]).each { |script| system!(env_vars, script) }
+        end
+
+        Array(travis_yml["after_script"]).each { |after_script| system!(env_vars, after_script) }
```

Comment:
I this might be clear to build up the list first and then execute it...something like

```suggestion
        commands =
          Array(travis_yml["before_install"]) +
          Array(travis_yml["install"]) +
          Array(travis_yml["before_script"]) +
          (script_cmd || Array(travis_yml["script"])) +
          Array(travis_yml["after_script"])
          
        commands.each { |command| system!(env_vars, command) }
```

Reply:
:+1: done